### PR TITLE
Ab1805 Setup with Return Codes

### PR DESCRIPTION
**Pull request by RadekPudelko**

### CODE REVIEW

File: src/AB1805_RK.h [hunk 1]
```diff
@@ -74,8 +74,11 @@ class AB1805 {
      * 
      * The FOUT/nIRQ pin is also used for one-time and periodic interrupts.
      */
-    AB1805 &withFOUT(pin_t pin) { foutPin = pin; return *this; };
-
+    AB1805 &withFOUT(pin_t pin) { 
+        foutPin = pin; 
+        pinMode(foutPin, OUTPUT);
```
Comment (josephfemia) [inline src/AB1805_RK.h:79]: Do we not already set the pin as an output in our pinModes or in other firmwares at the beginning of setup?

File: src/AB1805_RK.cpp [hunk 2]
```diff
@@ -39,12 +39,12 @@ void AB1805::setup(bool callBegin) {
 
             _log.info("set system clock from RTC %s", Time.format(time, TIME_FORMAT_DEFAULT).c_str());
         }
-    }
-    else {
+        System.on(reset, systemEventStatic);
```
Comment (josephfemia) [inline src/AB1805_RK.cpp:42]: Im fine w this but its interesting how if we failed to detect the chip we still wrote to registers right before a reset

**Review by josephfemia — CHANGES_REQUESTED** [review]

File: src/AB1805_RK.h (hunk 1 above)
Comment (RadekPudelko) [reply src/AB1805_RK.h:79]: Most of our drivers are setting pinModes internally, which I think is prefered

File: src/AB1805_RK.cpp (hunk 2 above)
Comment (RadekPudelko) [reply src/AB1805_RK.cpp:42]: yeah lol

**Review by josephfemia — APPROVED** [review]

**Pull request merged**